tests/test_dsl.py

```diff
@@ -320,3 +326,273 @@ def assert_cohorts_equivalent(dsl_cohort, qm_cohort):
     # ...and if the columns are the same.
     for k in dsl_col_defs:
         assert repr(dsl_col_defs[k]) == repr(qm_col_defs[k])
+
+
+def test_boolseries_and(bool_series):
+    series1 = bool_series()
+    series2 = bool_series()
+
+    expected = Comparator(
+        connector="and_",
+        negated=False,
+        operator=None,
+        lhs=Comparator(lhs=series1.value, operator="__ne__"),
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 & series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_boolseries_or(bool_series):
+    series1 = bool_series()
+    series2 = bool_series()
+
+    expected = Comparator(
+        connector="or_",
+        negated=False,
+        operator=None,
+        lhs=Comparator(lhs=series1.value, operator="__ne__"),
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 | series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_codeseries_and(code_series):
+    series1 = code_series()
+    series2 = code_series()
+
+    expected = Comparator(
+        connector="and_",
+        negated=False,
+        operator=None,
+        lhs=Comparator(lhs=series1.value, operator="__ne__"),
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 & series2
+    assert repr(output) == repr(BoolSeries(value=expected))
```

Comment:
What does `and`ing a `CodesSeries` mean?

Reply:
This is used in the categorising code, [example in the tests here](https://github.com/opensafely-core/databuilder/blob/ec0d400c187975f639f39653d83c7b2db164e8f1/tests/test_dsl_categorise.py#L331).